source/draw/RasterMonochrome.ooc

```diff
@@ -197,30 +197,35 @@ RasterMonochrome: class extends RasterPacked {
 
 	operator []= (x, y: Int, value: ColorMonochrome) {
 		version(safe) {
-			if (x > this size width || y > this size height || x < 0 || y < 0)
+			if (x >= this size width || y >= this size height || x < 0 || y < 0)
 				raise("Accessing RasterMonochrome index out of range in set operator")
 		}
 		((this buffer pointer + y * this stride) as ColorMonochrome* + x)@ = value
 	}
-	getRow: func (y: Int) -> FloatVectorList {
-		result := FloatVectorList new()
+	getRow: func (row: Int) -> FloatVectorList {
+		result := FloatVectorList new(this size width)
+		this getRowInto(row, result)
+		result
+	}
+	getRowInto: func (row: Int, vector: FloatVectorList) {
 		version(safe) {
-			if (y > this size height || y < 0)
+			if (row >= this size height || row < 0)
 				raise("Accessing RasterMonochrome index out of range in getRow")
 		}
-		for (x in 0 .. (this size width))
-				result add(this buffer pointer[y * this stride + x] as Float)
+		for (x in 0 .. this size width)
+				vector add(this buffer pointer[row * this stride + x] as Float)
```

Comment:
This is probably a good idea, but maybe in a separate pull request. I wrote these functions because the code I'm working on called 'getColumn' and 'getRow' a lot, allocating and destroying large buffers over and over again, causing really bad performance.